freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -64,4 +113,131 @@ public void deleteHost(HostRequest request) {
         //TODO  Will be implemented CDPSDX-515
         return;
     }
+
+    private FreeIpaClient getFreeIpaClient(Stack stack) {
+        String failure;
+        FreeIpaClient freeIpaClient = null;
+
+        try {
+            freeIpaClient = freeIpaClientFactory.getFreeIpaClientForStack(stack);
+        } catch (Exception e) {
+            failure = e.getLocalizedMessage();
+            LOGGER.warn("Failed to synchronize environment {}", stack.getEnvironmentCrn());
+            throw new RuntimeException("Failed to create IPA Client. " + failure);
+        }
+        // Using the freeIpaClient, hostname, serviceName and the realm we ca
+        return freeIpaClient;
+    }
+
+    private Stack getFreeIpaStack(String envCrn) {
+        FreeIpaClient freeIpaClient = null;
+        String failure;
+        String accountId = crnService.getCurrentAccountId();
+        LOGGER.debug("Looking for stack using env:{} and account-id:{}", envCrn, accountId);
+        List<Stack> stacks = stackService.findAllByEnvironmentCrnAndAccountId(envCrn, accountId);
+        if (stacks != null && stacks.size() > 0) {
+            return stacks.get(0);
+        } else {
+            return null;
+        }
+    }
+
+    private String getRealm(Stack stack) {
+        FreeIpa freeIpa = freeIpaService.findByStack(stack);
+        if ((freeIpa != null) && !Strings.isNullOrEmpty(freeIpa.getDomain())) {
```

Comment:
freeipa here can't be null as `freeIpaService.findByStack` will throw an exception if not found